virtual_rainforest/models/abiotic_simple/simple_regression.py

```diff
@@ -0,0 +1,726 @@
+r"""The ``models.abiotic_simple.simple_regression`` module uses linear regression
+and logarithmic interpolation to calculate atmospheric temperature and humidity profiles
+as a function of leaf area index and canopy layer height. The relationships are derived
+from :cite:t:`hardwick_relationship_2015` and :cite:t:`jucker_canopy_2018`.
+
+Soil temperature is interpolated between the surface layer and the air
+temperature at 1 m depth which equals the mean annual temperature.
+
+The module also provides a constant vertical profile of atmospheric pressure and
+:math:`\ce{CO2}`.
+
+Soil moisture and surface runoff are calculated with a simple bucket model based on
+:cite:t:`davis_simple_2017`.
+"""  # noqa: D205, D415
+
+from typing import Dict, List, Tuple, Union
+
+import numpy as np
+import xarray as xr
+from scipy.optimize import curve_fit
+from xarray import DataArray
+
+from virtual_rainforest.core.data import Data
+
+MicroclimateGradients: Dict[str, float] = {  # DOCSTRING? `hardwick_relationship_2015`
+    "air_temperature_gradient": -1.27,
+    "relative_humidity_gradient": 5.4,
+    "vapor_pressure_deficit_gradient": -252.24,
+}
+
+MicroclimateParameters: Dict[str, float] = {
+    "soil_moisture_capacity": 150,
+    "water_interception_factor": 0.1,
+    "saturation_vapor_pressure_factor1": 0.61078,
+    "saturation_vapor_pressure_factor2": 7.5,
+    "saturation_vapor_pressure_factor3": 237.3,
+}
+
+
+def setup_simple_regression(
+    layer_roles: List[str],
+    data: Data,
+    initial_soil_moisture: Union[DataArray, float] = 50,
+) -> List[DataArray]:
+    r"""Set up abiotic environment variables.
+
+    This function initialises all abiotic variables as DataArrays filled with NaN,
+    except for soil moisture which has a homogenous value across all soil layers.
+
+    Args:
+        layer_roles: list of layer roles (from top to bottom: above, canopy, subcanopy,
+            surface, soil)
+        data: Data object
+        initial_soil_moisture: initial soil moisture
+
+    Returns:
+        list of DataArrays for air temperature, relative humidity, vapor pressure
+        deficit, soil temperature, atmospheric pressure, atmospheric :math:`\ce{CO2}`,
+        soil moisture, and surface runoff
+    """
+
+    output = []
+
+    air_temperature = DataArray(
+        np.full((len(layer_roles), len(data.grid.cell_id)), np.nan),
+        dims=["layers", "cell_id"],
+        coords={
+            "layers": np.arange(0, len(layer_roles)),
+            "layer_roles": (
+                "layers",
+                layer_roles[0 : len(layer_roles)],
+            ),
+            "cell_id": data.grid.cell_id,
+        },
+        name="air_temperature",
+    )
+    """Mean air temperature profile, [C]"""
```

Comment:
I'm not sure if it makes sense to attach doc string to variables defined within a function, might be better just as a normal comment

Reply:
Yeah  - functions don't have attributes, so this docstring won't be attached to any object.

Reply:
Oh yes, of course. This is left from when I used to have classes and defined the attributes.